src/consensus.rs

```diff
@@ -491,6 +491,13 @@ impl Consensus {
             return Ok(None);
         }
 
+        if self
+            .try_add_origin()
+            .context("failed to propose origin peer URI to consensus")?
+        {
+            return Ok(Some(1));
+        }
```

Comment:
Shouldn't we just add the origin in the `init` function, for the non-bootstrap branch?

Reply:
(Discussed outside GH, posting comment here for the context.)

This was my initial idea: to simply push a single `AddPeer` message to consensus channel, even before starting consensus thread/loop, so that it will be processed once consensus started. But unfortunately, it's never that easy with Raft. 😭

When origin node is started for the first time (i.e., when you start a new cluster), it needs to finish a self-election before you can propose/commit anything to consensus (it will return an error before leader is elected). So you need to spin consensus loop few times (to tick Raft node and process election) before you can propose and commit origin peer.

So I decided to organize `try_add_origin` a bit like `try_promote_learner` and add it as part of the consensus loop.